### PR TITLE
testing/ssh-honeypot: new aport

### PR DESCRIPTION
**Pull request by uu**
This program listens for incoming ssh connections and logs the ip address, username, and password used. This was written to gather rudimentary intelligence on brute force attacks.

https://github.com/droberson/ssh-honeypot

### CODE REVIEW

File: testing/ssh-honeypot/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Contributor: Michael Pirogov <vbnet.ru@gmail.com>
+# Maintainer: Michael Pirogov <vbnet.ru@gmail.com>
+pkgname="ssh-honeypot"
+pkgver="0.1.0"
```
Comment (Cogitri) [inline testing/ssh-honeypot/APKBUILD]: should be 0_git<date of snapshot>

File: testing/ssh-honeypot/10-Makefile.patch [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
++++ b/Makefile
+@@ -1,9 +1,10 @@
+ CC=clang
+ CFLAGS=-Wall -static-libgcc
+ LIBS=-lssh -ljson-c
++DESTDIR:=
+ 
+ ssh-honeypot:
+-	$(CC) $(CFLAGS) -o bin/ssh-honeypot src/ssh-honeypot.c $(LIBS)
++	$(CC) $(CFLAGS) -o $(DESTDIR)/bin/ssh-honeypot src/ssh-honeypot.c $(LIBS)
```
Comment (Cogitri) [inline testing/ssh-honeypot/10-Makefile.patch]: Seems weird to build and install here, just remove this patch and install in the APKBUILD

File: testing/ssh-honeypot/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+subpackages="$pkgname-openrc"
+source="$pkgname-$pkgver.zip::https://github.com/droberson/ssh-honeypot/archive/$_commit.zip
+	$pkgname.initd
+	$pkgname.confd"
+builddir="$srcdir/$pkgname-$_commit"
+
+
+package() {
+	mkdir -p $pkgdir/usr/bin/ $pkgdir/usr/share/ssh-honeypot
+	make PREFIX=/usr DESTDIR="$pkgdir" -C "$builddir"
```
Comment (Cogitri) [inline testing/ssh-honeypot/APKBUILD]: Just remove this line

File: testing/ssh-honeypot/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	install -m 755 -D bin/$pkgname $pkgdir/usr/bin/$pkgname
```
Comment (Cogitri) [inline testing/ssh-honeypot/APKBUILD]: and do `install -m 755 -D "$builddir"/bin/$pkgname "$pkgdir"/usr/bin/$pkgname`

File: testing/ssh-honeypot/APKBUILD [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+	install -m 755 -D "$srcdir"/$pkgname.initd $pkgdir/etc/init.d/$pkgname
+	install -m 644 -D "$srcdir"/$pkgname.confd $pkgdir/etc/conf.d/$pkgname
```
Comment (Cogitri) [inline testing/ssh-honeypot/APKBUILD]: quote $pkgname too (it's possible that it contains a space)

File: testing/ssh-honeypot/APKBUILD [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+options="!check"
+subpackages="$pkgname-openrc"
+source="$pkgname-$pkgver.zip::https://github.com/droberson/ssh-honeypot/archive/$_commit.zip
+	$pkgname.initd
+	$pkgname.confd"
+builddir="$srcdir/$pkgname-$_commit"
+
+
+package() {
+	mkdir -p $pkgdir/usr/bin/ $pkgdir/usr/share/ssh-honeypot
```
Comment (Cogitri) [inline testing/ssh-honeypot/APKBUILD]: same here, please quote

File: testing/ssh-honeypot/APKBUILD [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+pkgname="ssh-honeypot"
+pkgver="0.1.0"
+_commit=b7baa9a6963e723ee0779a6cef0447739870ace3
+pkgrel=0
+pkgdesc="Fake sshd that logs ip addresses, usernames, and passwords."
+url="https://github.com/droberson/ssh-honeypot"
+arch="all"
+license="MIT"
+depends="openssh-keygen"
+makedepends="clang json-c0.12-dev libssh-dev"
```
Comment (Cogitri) [inline testing/ssh-honeypot/APKBUILD]: Why the dependency on clang?

File: testing/ssh-honeypot/ssh-honeypot.initd [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+key_path="/usr/share/${prog_name}/honeypot.rsa"
+pid_file="/var/run/${prog_name}.pid"
+start_stop_daemon_args="-b -m --pidfile ${pidfile} -- "
+
+depend() {
+  need net
+}
+
+start_pre() {
+  if [[ ! -e "${key_path}" ]]; then
```
Comment (Cogitri) [inline testing/ssh-honeypot/ssh-honeypot.initd]: I think you'd usually use checkpath in initd scripts

File: testing/ssh-honeypot/APKBUILD (hunk 5 above)
Comment (uu) [reply testing/ssh-honeypot/APKBUILD]: I think you meant $pkgdir ;)

File: testing/ssh-honeypot/APKBUILD (hunk 7 above)
Comment (uu) [reply testing/ssh-honeypot/APKBUILD]: replaced with gcc

File: testing/ssh-honeypot/APKBUILD (hunk 5 above)
Comment (Cogitri) [reply testing/ssh-honeypot/APKBUILD]: Uhh...right :P

File: testing/ssh-honeypot/APKBUILD (hunk 7 above)
Comment (Cogitri) [reply testing/ssh-honeypot/APKBUILD]: That should be in there by default, so no need to depend on that.

File: testing/ssh-honeypot/APKBUILD [hunk 9]
```diff
@@ -0,0 +1,32 @@
+# Contributor: Michael Pirogov <vbnet.ru@gmail.com>
+# Maintainer: Michael Pirogov <vbnet.ru@gmail.com>
+pkgname="ssh-honeypot"
+pkgver="0_git"
```
Comment (Cogitri) [inline testing/ssh-honeypot/APKBUILD]: You forgot to append the date to this (see other git packages which do this)

File: testing/ssh-honeypot/APKBUILD [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+subpackages="$pkgname-openrc"
+source="$pkgname-$pkgver.zip::https://github.com/droberson/ssh-honeypot/archive/$_commit.zip
+	$pkgname.initd
+	$pkgname.confd"
+builddir="$srcdir/$pkgname-$_commit"
+
+
+package() {
+	mkdir -p "$pkgdir"/usr/bin/ "$pkgdir"/usr/share/ssh-honeypot
+	gcc -Wall -static-libgcc -o "$builddir"/bin/$pkgname "$builddir"/src/ssh-honeypot.c -lssh -ljson-c
```
Comment (Cogitri) [inline testing/ssh-honeypot/APKBUILD]: Do this in build please and use cc instead gcc. Also, can't the makefile do this?

File: testing/ssh-honeypot/APKBUILD (hunk 7 above)
Comment (Cogitri) [reply testing/ssh-honeypot/APKBUILD]: Ah you did that, my bad

File: testing/ssh-honeypot/APKBUILD (hunk 10 above)
Comment (uu) [reply testing/ssh-honeypot/APKBUILD]: Yep, moved back to the Makefile.

File: testing/ssh-honeypot/APKBUILD [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+makedepends="json-c0.12-dev libssh-dev"
+options="!check"
+subpackages="$pkgname-openrc"
+source="$pkgname-$pkgver.zip::https://github.com/droberson/ssh-honeypot/archive/$_commit.zip
+	$pkgname.initd
+	$pkgname.confd"
+builddir="$srcdir/$pkgname-$_commit"
+
+build() {
+	make CC='gcc -fPIC' || return 1
```
Comment (Cogitri) [inline testing/ssh-honeypot/APKBUILD]: Do `CFLAGS="$CFLAGS -fPIC` instead. Also, no need for `|| return 1`, APKBUILDs automatically fail if one of the commands returns != 0

File: testing/ssh-honeypot/APKBUILD [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+pkgver="0_git20190717"
+_commit=b7baa9a6963e723ee0779a6cef0447739870ace3
+pkgrel=0
+pkgdesc="Fake sshd that logs ip addresses, usernames, and passwords."
+url="https://github.com/droberson/ssh-honeypot"
+arch="all"
+license="MIT"
+depends="openssh-keygen"
+makedepends="json-c0.12-dev libssh-dev"
+options="!check"
```
Comment (Cogitri) [inline testing/ssh-honeypot/APKBUILD]: Please add a comment explaining why they're disabled (e.g. `# no tests`)

File: testing/ssh-honeypot/APKBUILD (hunk 11 above)
Comment (uu) [reply testing/ssh-honeypot/APKBUILD]: Snitched that from community/snowball;)

File: testing/ssh-honeypot/APKBUILD [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+makedepends="json-c0.12-dev libssh-dev"
+options="!check" # no testsuite and daemon returns 1 even on help command
+subpackages="$pkgname-openrc"
+source="$pkgname-$pkgver.zip::https://github.com/droberson/ssh-honeypot/archive/$_commit.zip
+	$pkgname.initd
+	$pkgname.confd"
+builddir="$srcdir/$pkgname-$_commit"
+
+build() {
+	make CC="gcc" CFLAGS="$CFLAGS -fPIC"
```
Comment (Cogitri) [inline testing/ssh-honeypot/APKBUILD]: No need to set `CC` now, that should be set to `CC` by default.

**Comment by Cogitri** [comment]
LGTM otherwise.

File: testing/ssh-honeypot/APKBUILD (hunk 13 above)
Comment (uu) [reply testing/ssh-honeypot/APKBUILD]: I have to because of CC=clang in the Makefile.

Comment (Cogitri) [reply testing/ssh-honeypot/APKBUILD]: Sorry about that, you're right. Had thought our env variable would take precedence here (since we export `CC`)

**Pull request closed without merging**